btree/btree.c

```diff
@@ -3596,21 +3604,24 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 		if (!path_check(oi, tree, &bop->bo_rec.r_key.k_cookie)) {
 			oi->i_trial++;
 			if (oi->i_trial >= MAX_TRIALS) {
-				if (bop->bo_flags & BOF_LOCKALL) {
-					lock_op_unlock(bop->bo_arbor->t_desc);
-					return fail(bop, -ETOOMANYREFS);
-				} else
-					bop->bo_flags |= BOF_LOCKALL;
+				M0_ASSERT_INFO((bop->bo_flags & BOF_LOCKALL) ==
```

Comment:
Do you need this ASSERT or you forgot to remove from here. Please check once

Reply:
We need it. Suppose we have already set LOCKALL and still path check fails then there is something seriously wrong with code.